### DISCUSSION TITLE
How to access tasmota file system?  (don't see it in the GUI)

### DISCUSSION THREAD

**Discussion by Sum-Guy** [arendst/Tasmota#17566] [Q&A] [answered]
I'm using the benzino tasmocomiler and the option "SD card/LittleFS" is checked.

-----------
After compiling and flashing you will find a new entry in Tasmota webUI: Consoles - Manage File system
------------

I'm not seeing that.  The example seems to show a button labelled as "Manage File System" under the "Restore Configuration" button, which would be in the Configuration menu, but I'm not seeing it.

Is there a command I can give in the console to verify that a file system is present on the device?

Or does some other compiler directive need to be set to have the file system in the build?

"This feature is included in every precompiled binary with flash size >1M"

Honestly, I've never seen the "Manage File System" GUI button and I've putzed around alot with these things...

Program Version	12.1.1(TasmoCompiler-esp82664M)

user config override file contains:

#ifdef USE_UFILESYS
  #undef USE_UFILESYS
#endif
#define USE_UFILESYS

#ifdef USE_SDCARD
  #undef USE_SDCARD
#endif
#define USE_SDCARD

#ifdef GUI_TRASH_FILE
  #undef GUI_TRASH_FILE
#endif
#define GUI_TRASH_FILE

#ifdef GUI_EDIT_FILE
  #undef GUI_EDIT_FILE
#endif
#define GUI_EDIT_FILE

Is this correct code?

**Comment by Sum-Guy** [comment]
Is this why I don't see it?  Because the "program flash size" is 1 mb?

ESP Chip Id	10352055 (ESP8266EX)
Flash Chip Id	0x164020
Flash Size	4096 KB
Program Flash Size	1024 KB
Program Size	757 KB
Free Program Space	1288 KB
Free Memory	24.0 KB

**Reply by barbudor** [reply]
In Tasmocompiler you should be able to select a "board" where you should be able to select an appropriate configuration. Default is for a 1MB ESP so no room for file system. 
I'm not at a computer right now but I can have a look tonight if you can't find it

**Reply by Sum-Guy** [reply]
The ESP8266 board I select is Wemos/NodeMCU.  The SD card/LittleFS is automatically selected.

The info page on tasmota web gui says:

Flash Size 4096 KB
Program Flash Size 1024 KB

What is "program flash size" ? And can it be / should it be set to 4mb?  And if so, how is that done?

I've worked with ESP8266 with 1mb and 4 mb (and prob. 512kb) flash and ESP32, usually with the pre-compiled binaries, and I've never seen the "manage file system" in the web GUI.   I get it that it technically only shows up on units with 4 mb flash, but I have both 1 and 4 mb devices.

Side question:  Did or does any ESP32 board ever come with 1 mb flash?  Or is it assumed that any ESP32 board will have 4mb?

**Comment by Jason2866** [comment]
`pio run -e tasmota4M` should be enough.

**Reply by Sum-Guy** [reply]
Using the benzino tasmocomiler -

In the platformio_override.ini file (which I have not modified) these are the only entries:

[platformio]
default_envs = tasmota4M

[env:tasmota4M]
board = esp8266_4M2M


->  "pio run -e tasmota4M"

I don't think the tasmocompiler gives me direct control over the pio run argument.

**Reply by Jason2866** [reply]
Ahh TasmoCompiler, idk better to ask there how to use an existing Tasmota env

**Reply by Jason2866** [reply]
The variant NodeMCU uses 4MB linker script so the file system is enabled by default

**Reply by barbudor** [reply]
Just tested from the development branch, and it's working fine
How did you flashed ?
You MUST fully erase the flash before flashing a firmware with filesystem support. Upgrade doesn't work. So serial flashing is required

**Reply by Sum-Guy** [reply]
I'm flashing using tasmotizer.  I have 2 executables:

tasmotizer_x86-1.1b.exe
tasmotizer_x86-1.1c.exe

I don't know what the difference is between them, they are dated Sept 11 and 12, 2020.  I've been using the "b" version.   It has a check-box "erase before flashing" and its always checked.

The .bin file that is generated by tasmocompiler is named "tasmota4M.bin" (but it's 758kb in size).

Would the size of the .bin tell us anything about whether it's designed for a 4M vs 1M flash?

Why am I seeing this:

Program Flash Size 1024 KB

On the Information screen on the web GUI?   Does that give a clue as to what's going on?  Should it say 4096 KB?

**Reply by Sum-Guy** [reply]
Is anything here relavent?
https://github.com/arendst/Tasmota/issues/1728
Program Flash Size vs Flash Size

Or here:
https://github.com/arendst/Tasmota/pull/10528
Support for 2M and 4MB builds

**Comment by Sum-Guy** [comment, marked as answer]
I've just downloaded what looks like the most current version of Tasmotizer:

tasmotizer_x86-1.2.exe 

Which seems to have been created Sept 21 / 2020.  What I was using was tasmotizer_x86-1.1b.exe (Sept 12 / 2020).

Using v 1.2 seems to have made a difference.  Flashing the exact same tasmota4M.bin file I now see this:

Flash Size	4096 KB
Program Flash Size	4096 KB

and I also see this from the serial logging port on device boot:

UFS: FlashFS mounted with 1992 kB free

And yes, I now see "Consoles" (instead of Console) on the main web GUI page, and yes in there I now see "Manage File System".

So, the older version of Tasmotizer seems to have been the problem, no idea why.  Just don't use it I guess is the answer here.

**Reply by barbudor** [reply]
The size of the file is ok. Created for 1M or 4M just means it knows what is the flash boundaries. But Tasmota firmware itself doesn't change it's size and remains in the lower 1MB of the flash, leaving the rest for the FileSystem
I am not ure that Tasmotizer is able to flash 4M targets. I never used it for that. May be @jziolkowski tell us

For flashing 4M targets, I'm using the following command line with ESPTOOL:
```
esptool --port $1 write_flash -fs 4MB -fm dout 0x0 $2
```
where `$1` is the name of the port (like COM3, /dev/ttyUSB0, ..) and `$2` is the name of the binary file (tasmota4M.bin in your case)

**Reply by sfromis** [reply]
Tasmotizer is no longer the recommended tool for flashing (unless the creator does some work on it), the Tasmota Web Installer generally has better support for variations.

**Reply by barbudor** [reply]
But the webinstaller doesn't allow to flash custom built binaries
So you have to 1st flash the "tasmota for 4MB" and then upgrade to your own binary....
esptool still the simplest IMHO for those cases

**Reply by sfromis** [reply]
I was mainly trying to point out that Tasmotizer is not really recommended now

**Reply by jziolkowski** [reply]
Tasmotizer doesn't set the bin size. It only sets DOUT mode, and starting address 0x0. Then esptool takes care of detecting the flash size
